lib/oli_web/components/delivery/scored_activities/scored_activities.ex

```diff
@@ -483,81 +418,45 @@ defmodule OliWeb.Components.Delivery.ScoredActivities do
 
   defp maybe_filter_by_text(assessments, text_search) do
     Enum.filter(assessments, fn assessment ->
-      String.contains?(
-        String.downcase(assessment.title),
-        String.downcase(text_search)
-      )
+      String.contains?(String.downcase(assessment.title), String.downcase(text_search))
     end)
   end
 
   defp sort_by(assessments, sort_by, sort_order) do
     case sort_by do
       :due_date ->
-        Enum.sort_by(
-          assessments,
-          fn a ->
-            if a.scheduling_type != :due_by, do: 0, else: Map.get(a, :due_date)
-          end,
-          sort_order
-        )
+        Enum.sort_by(assessments, & &1.end_date, sort_order)
```

Comment:
I think this fix would sort by end date instead of doing it by due date. My understanding is that we only want to consider assessments with `scheduling_type == :due_by` in the sorting (and group all other cases in one same "fake date"). 
I think the following code will work:
```
     Enum.sort_by(
          assessments,
          &if(&1.scheduling_type != :due_by, do: 0, else: Map.get(&1, :end_date)),
          sort_order
        )
```
Note that this code is almost the same as the original one, but I fixed it to get `:end_date` of the assessment instead of `:due_date` (that key does not exist in the map)

## Before

https://github.com/Simon-Initiative/oli-torus/assets/74839302/7b9658e4-f725-4815-9d7e-c7e7074fb9bb



## After

https://github.com/Simon-Initiative/oli-torus/assets/74839302/29610ebd-9467-4f50-a0af-7c04935c70a7

Reply:
Thanks @nicocirio for the suggestion and the explanation. I can see now that the Due Date column depends on two values: `scheduling_type` and the `section_resource.end_date`. I'll do that changes then and let you know.